pkg/server/tls_windows.go

```diff
@@ -0,0 +1,342 @@
+package server
+
+import (
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/asn1"
+	"fmt"
+	"regexp"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/github/smimesign/certstore"
+)
+
+func (l *tlsListener) applyWindowsCertificateStore(c TLSConfig) error {
+
+	// Restrict normal TLS options when using windows certificate store
+	if c.TLSCertPath != "" {
+		return fmt.Errorf("cannot include certificate file when using windows certificate store")
+	}
+	if c.TLSKeyPath != "" {
+		return fmt.Errorf("cannot include key file when using windows certificate store")
+	}
+
+	var subjectRegEx *regexp.Regexp
+	var err error
+	if c.WindowsCertificateFilter.ClientSubjectRegEx != "" {
+		subjectRegEx, err = regexp.Compile(c.WindowsCertificateFilter.ClientSubjectRegEx)
+		if err != nil {
+			return fmt.Errorf("error compiling subject common name regular expression %w", err)
+		}
+	}
+
+	// If there is an existing windows certhandler notify it to stop refreshing
+	if l.windowsCertHandler != nil {
+		l.windowsCertHandler.stopUpdateTimer <- struct{}{}
+	}
+
+	cn := &winCertStoreHandler{
+		cfg:             *c.WindowsCertificateFilter,
+		subjectRegEx:    subjectRegEx,
+		stopUpdateTimer: make(chan struct{}, 1),
+	}
+	err = cn.refreshCerts()
+	if err != nil {
+		return err
+	}
+	certPool := x509.NewCertPool()
+	certPool.AddCert(cn.clientRootCA)
+	config := &tls.Config{
+		VerifyPeerCertificate: cn.VerifyPeer,
+		GetCertificate: func(info *tls.ClientHelloInfo) (*tls.Certificate, error) {
+			cn.winMut.Lock()
+			defer cn.winMut.Unlock()
+			cert := &tls.Certificate{
+				Certificate: [][]byte{cn.serverCert.Raw},
+				PrivateKey:  cn.serverSigner,
+				// These seem to the be safest to use, tested on Win10, Server 2016, 2019, 2022
+				SupportedSignatureAlgorithms: []tls.SignatureScheme{
+					tls.PKCS1WithSHA256,
+					tls.PKCS1WithSHA384,
+					tls.PKCS1WithSHA512,
+				},
```

Comment:
I'm not sure I understand why we need to provide this list yet. It restricts which algorithms are used, so if it's not provided, everything is valid, right?